cupy/cuda/graph.pyx

```diff
@@ -78,3 +77,25 @@ cdef class Graph:
         else:
             stream_ptr = stream.ptr
         runtime.graphUpload(self.graphExec, stream_ptr)
+
+    cpdef debug_dot_str(self, flags=0):
+        """Make DOT formatted string of CUDA graph definition for debugging.
+
+        Args:
+            flags (:class:`unsigned int`): Flags to specify information to be
+                included.
+
+        .. seealso:: `cudaGraphDebugDotPrint()`_
+
+        .. _cudaGraphDebugDotPrint():
+            https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__GRAPH.html#group__CUDART__GRAPH_1gbec177c250000405c570dc8c4bde20db
+
+        """
+        f = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            f.close()
```

Comment:
Q: Why is `f` closed prior to accessing it?

Reply:
(It's added in my follow-up PR: https://github.com/so298/cupy/pull/2 😄)

Windows can't open the same file twice, so we need to close it before passing the path to CUDA runtime.